### ISSUE TITLE
False positive nonlocal-without-binding for class defined inside function

### ISSUE THREAD

**Issue by mwchase** [PyCQA/pylint#1574]
### Code to reproduce
```
def func():
    nonlocal_ = None
    local = None

    class Class:

        nonlocal nonlocal_

        nonlocal_ = 1
        local = 1

    print(nonlocal_)
    print(local)


func()

```

### Current behavior
Code prints
```
1
None
```
Pylint produces several complaints, including `E:  7, 8: nonlocal name nonlocal_ found without binding (nonlocal-without-binding)`

### Expected behavior
Pylint should not generate that error.

### pylint --version output
```
No config file found, using default configuration
pylint 1.7.2,
astroid 1.5.3
Python 3.6.1 (default, Mar 23 2017, 16:49:06)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

**Comment by PCManticore** [comment]
I can reproduce this, thanks!

What is happening is that the checker expects the nonlocals to be present in a FunctionDef scope, not a class scope. This should be an easy fix.
